numpy/core/src/multiarray/_multiarray_tests.c.src

```diff
@@ -176,17 +176,21 @@ test_neighborhood_iterator(PyObject* NPY_UNUSED(self), PyObject* args)
     /* Compute boundaries for the neighborhood iterator */
     for (i = 0; i < 2 * PyArray_NDIM(ax); ++i) {
         PyObject* bound;
+        npy_intp tmp;
+
         bound = PySequence_GetItem(b, i);
         if (bound == NULL) {
             goto clean_itx;
         }
-        if (!PyInt_Check(bound)) {
+        /* PyLong_AsSsize checks for PyLong */
+        if ((tmp = PyLong_AsSsize_t(bound)) == -1 && PyErr_Occurred()) {
+            PyErr_Clear();
```

Comment:
Since this is just test code, I'd be inclined to throw out this custom error handling and just propagate the exception that is already set

Reply:
Seems reasonable.